File: src/main/java/edu/univ/erp/api/auth/AuthApi.java
package edu.univ.erp.api.auth;

import edu.univ.erp.auth.AuthService;
import org.springframework.stereotype.Service;

@Service
public class AuthApi {

    private final AuthService authService;

    public AuthApi(AuthService authService) {
        this.authService = authService;
    }

    public LoginResult attemptLogin(String username, String password) {
        return authService.login(username, password);
    }

    public boolean changePassword(int userId, String currentPassword, String newPassword) {
        return authService.changePassword(userId, currentPassword, newPassword);
    }
}

File: erp-web/lib/api/auth_api.dart
import 'client.dart';

class AuthApi {
  static Future<Map<String, dynamic>> login(
      String username, String password) async {
    final res = await apiClient
        .post('/auth/login', data: {'username': username, 'password': password});
    return res.data as Map<String, dynamic>;
  }

  static Future<void> changePassword(
      String currentPassword, String newPassword) async {
    await apiClient.post('/auth/change-password',
        data: {'currentPassword': currentPassword, 'newPassword': newPassword});
  }
}
